### PR TITLE
refactor: remove std txn from domain

### PR DESCRIPTION
**Pull request by SimonRichardson**
We shouldn't be using StdTxn from inside the domain package. It's fine with in tests, but all queries should be using sqlair.

Reducing the interface of the TxnRunner then allows to add a TxnWithPrecheck to prevent bloating out the TxnRunner. We want to clean up and ensure clarity whilst we go.

## Checklist

- [ ] Code style: imports ordered, good names, simple structure, etc
- [ ] Comments saying why design decisions were made
- [ ] Go unit tests, with comments saying what you're testing
- [ ] [Integration tests](https://github.com/juju/juju/tree/main/tests), with comments saying what you're testing
- [ ] [doc.go](https://discourse.charmhub.io/t/readme-in-packages/451) added or updated in changed packages

## Links

**Launchpad bug:** https://bugs.launchpad.net/juju/+bug/

**Jira card:** JUJU-

### CODE REVIEW

File: domain/credential/state/state.go [hunk 1]
```diff
@@ -443,58 +443,50 @@ type credentialAttrs map[string]string
 // LoadCloudCredentials loads cloud credentials from the database.
 func LoadCloudCredentials(ctx context.Context, st domain.Preparer, tx *sqlair.TX, name, cloudName string, owner user.Name) ([]credential.CloudCredentialResult, error) {
 	credQuery := `
-SELECT (cc.uuid, cc.name,
-       cc.revoked, cc.invalid, 
-       cc.invalid_reason, 
-       cc.owner_uuid) AS (&Credential.*),
+SELECT &Credential.*,
        auth_type.type AS &authType.type,
-       cloud.name AS &dbCloudName.name,
```
Comment (Aflynn50) [inline domain/credential/state/state.go:451]: No need to get the cloud name out of the DB when its a query parameter

File: domain/credential/state/state.go [hunk 2]
```diff
@@ -386,23 +386,62 @@ AND    cloud_credential.cloud_uuid = (
...
 
-	var creds []credential.CloudCredentialResult
+	var (
+		dbRows      Credentials
+		dbAuthTypes []authType
+		keyValues   []CredentialAttribute
+	)
 	err = db.Txn(ctx, func(ctx context.Context, tx *sqlair.TX) error {
-		var err error
-		creds, err = LoadCloudCredentials(ctx, st, tx, "", cloudName, owner)
```
Comment (Aflynn50) [inline domain/credential/state/state.go:400]: LoadCloudCredentials was overloaded to have 3 different functions based on the inputs. It relied on the (now deleted) `SqlairAndClause` to add conditions to the WHERE clause if the arguments were none zero.

I have split it into different functions to capture the different use cases, hopefully making it clear to readers of the code what is going on.

**Comment by hpidcock** [comment]
Handled in #18164

**Comment by SimonRichardson** [comment]
Not quite.

**Comment by Aflynn50** [comment]
> Not quite. That was for the model domain. We will need to finish all the other domains.

I'm putting up PRs for the others too, rather than doing it in this, see #18182

**Pull request closed without merging**